### PR TITLE
Simplify `F.leaky_relu` test

### PR DESCRIPTION
**Pull request by aksub99**
Simplified F.leaky_relu test using testing.FunctionTestCase.
Ref. #6071

### CODE REVIEW

**Comment by aksub99** [comment]
@Crissman Please review the PR and if there are any changes I can make, do let me know :)

File: tests/chainer_tests/functions_tests/activation_tests/test_leaky_relu.py [hunk 1]
```diff
@@ -47,56 +41,20 @@ def setUp(self):
...
-        if backend_config.use_cuda:
-            x_data = cuda.to_gpu(x_data)
-            y_grad = cuda.to_gpu(y_grad)
-            x_grad_grad = cuda.to_gpu(x_grad_grad)
-
-        def f(x):
-            return functions.leaky_relu(x, self.slope)
+    def generate_inputs(self):
+        x = numpy.random.uniform(-1, 1, self.shape).astype(self.dtype)
+        if self.slope == 'random':
```
Comment (Crissman) [inline tests/chainer_tests/functions_tests/activation_tests/test_leaky_relu.py]: To keep similar terms together, better to leave these `self.slope` lines in setUp.

**Comment by Crissman** [comment]
Otherwise, looks good!

**Comment by aksub99** [comment]
Thanks for reviewing @Crissman ! I have made the required changes.

**Comment by Crissman** [comment]
Jenkins, test this please.

**Comment by Crissman** [comment]
Jenkins failures look unrelated. Merging.

**Pull request merged**